helpers/metrics.py

```diff
@@ -176,25 +168,27 @@ def get_metrics(tp_gdf, fp_gdf, fn_gdf, mismatch_gdf, id_classes=0, method='macr
         fn_count = pure_fn_count + mismatched_fn_count
         tp_count = 0 if tp_gdf.empty else len(tp_gdf[tp_gdf.det_class==id_cl])
 
-        tp_k[id_cl] = tp_count
         fp_k[id_cl] = fp_count
         fn_k[id_cl] = fn_count
-    
-        p_k[id_cl] = 0 if tp_count == 0 else tp_count / (tp_count + fp_count)
-        r_k[id_cl] = 0 if tp_count == 0 else tp_count / (tp_count + fn_count)
-        count_k[id_cl] = 0 if tp_count == 0 else tp_count + fn_count 
+        tp_k[id_cl] = tp_count
+
+        count_k[id_cl] = tp_count + fn_count
+        if tp_count > 0:
+            p_k[id_cl] = tp_count / (tp_count + fp_count)
+            r_k[id_cl] = tp_count / (tp_count + fn_count)
+
+        if (method == 'macro-weighted-average') & (total_labels > 0):
+            pw_k[id_cl] = (count_k[id_cl] / total_labels) * p_k[id_cl]
+            rw_k[id_cl] = (count_k[id_cl] / total_labels) * r_k[id_cl] 
 
     if method == 'macro-average':   
         precision = sum(p_k.values()) / len(id_classes)
         recall = sum(r_k.values()) / len(id_classes)
-    elif method == 'macro-weighted-average': 
-        for id_cl in id_classes:
-            pw_k[id_cl] = 0 if sum(count_k.values()) == 0 else (count_k[id_cl] / sum(count_k.values())) * p_k[id_cl]
-            rw_k[id_cl] = 0 if sum(count_k.values()) == 0 else (count_k[id_cl] / sum(count_k.values())) * r_k[id_cl] 
+    elif method == 'macro-weighted-average':  
         precision = sum(pw_k.values()) / len(id_classes)
         recall = sum(rw_k.values()) / len(id_classes)
     elif method == 'micro-average':  
-        if sum(tp_k.values()) == 0 and sum(fp_k.values()) == 0:
```

Comment:
Don't we want to keep the previous version? When the test is about the denominateur being 0?
Because with `sum(tp_k.values()) ==0` one get in any case recall == precision ==0